src/reanimated2/shareables.ts

```diff
@@ -146,6 +146,8 @@ export function makeShareableCloneRecursive<T>(
         // then recreate new host object wrapping the same instance on the UI thread.
         // there is no point of iterating over keys as we do for regular objects.
         toAdapt = value;
+      } else if (isPlainJSObject(value) && global._hasNativeState(value)) {
```

Comment:
Why do we need this?

Reply:
`ShadowNodeWrapper` used to be a `jsi::HostObject` but now it's a regular JS object with `jsi::NativeState`. Currently, Reanimated does not propagate native state (only copies the keys/values of JS objects). This change is necessary to support RN 0.74+, however I will try to improve the implementation here.